components/Village/Edit/index.vue

```diff
@@ -0,0 +1,429 @@
+          :error-message="errors.city"
+          placeholder="Pilih Kabupaten/Kota"
+        />
+      </div>
+      <div class="form-edit-village__form-group">
+        <jds-select
+          v-model="form.district_id"
+          :disabled="isDisabledOptionDistricts"
+          class="form-edit-village__form-group-field--label"
+          name="Kecamatan"
+          filterable
+          label="Kecamatan"
+          options-header="Kecamatan"
+          :options="optionsDistrict"
+          :error-message="errors.district"
+          placeholder="Pilih Kecamatan"
+        />
+      </div>
+      <div class="form-edit-village__form-group">
+        <jds-select
+          v-model="form.level"
+          class="form-edit-village__form-group-field--label"
+          name="Level Desa"
+          label="Level Desa"
+          :options="optionsLevel"
+          placeholder="Pilih Level"
+          :error-message="errors.villageLevel"
+        />
+      </div>
+      <div class="form-edit-village__form-group form-edit-village__form-group--position">
+        <div class="w-[250px]">
+          <BaseInputText
+            v-model="form.longitude"
+            class="form-edit-village__form-group-field--label"
+            name="longitude"
+            label="Longitude"
+            type="number"
+            placeholder="Contoh: 6.219212"
+            autocomplete="false"
+            :error="!!(errors.longitude)"
+          />
+          <div v-if="errors.longitude" class="text-red-700">
+            {{ errors.longitude }}
+          </div>
+        </div>
+        <div class="w-[250px]">
+          <BaseInputText
+            v-model="form.latitude"
+            class="form-edit-village__form-group-field--label"
+            name="latitude"
+            label="Latitude"
+            type="number"
+            placeholder="Contoh: 6.219212"
+            autocomplete="false"
+            :error="!!(errors.latitude)"
+          />
+          <div v-if="errors.latitude" class="text-red-700">
+            {{ errors.latitude }}
+          </div>
+        </div>
+      </div>
+    </form>
+  </BaseModal>
+</template>
+
+<script>
+import { isEqual, cloneDeep } from 'lodash'
+export default {
+  name: 'ComponentVillageEdit',
+  props: {
+    /**
+     * make modal visible or not
+     */
+    show: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * object of village
+     */
+    item: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+  data () {
+    return {
+      loading: false,
+      form: {
+        id: '',
+        name: '',
+        city_id: '',
+        district_id: '',
+        level: 0,
+        longitude: '',
+        latitude: ''
+      },
+      errors: {
+        villageId: null,
+        villageName: null,
+        villageLevel: null,
+        city: null,
+        district: null,
+        longitude: null,
+        latitude: null
+      },
+      listCity: [],
+      listDistrict: [],
+      optionsLevel: [
```

Comment:
nice to have:
since `optionsLevel` is used a couple times (in add and edit) maybe we can put it into constants variable in `dataVillage.js`. can refactor it later..

Reply:
Noted 🚀